proof/courier.go

```diff
@@ -832,14 +836,15 @@ func (h *HashMailCourier) DeliverProof(ctx context.Context,
 		defer cancel()
 		err = h.mailbox.RecvAck(ctxTimeout, receiverStreamID)
 		if err != nil {
-			return fmt.Errorf("failed to receive ACK "+
+			return nil, fmt.Errorf("failed to receive ACK "+
 				"from receiver within timeout: %w", err)
 		}
 
-		return nil
+		return nil, nil
```

Comment:
The delivery function did not need to return the proof blob that was delivered. I've reformulated the "target function" callback in this PR's latest commits. If there is a proof blob that needs to be obtained from the closure we can use a closure variable rather than a returned variable. So this "returning nil for `[]byte` situation" has been resolved as well.